content/My favorite extensions for programming Ruby in Visual Studio Code.md

```diff
@@ -0,0 +1,106 @@
+
+I love Gitlens because it permits me to detect who wrote the code that I am reading, this is really useful because when I have doubts about the code, I know who to talk to.
+
+**Features:**
+
+- An unobtrusive current line blame annotation at the end of the line with detailed blame information accessible via hovers
+- Authorship code lens showing the most recent commit and # of authors to the top of files and/or on code blocks
+- A status bar blame annotation showing author and date for the current line
+- On-demand gutter blame annotations, including a heatmap, for the whole file
+- On-demand gutter heatmap annotations to show how recently lines were changed, relative to all the other changes in the file and to now (hot vs. cold)
+- On-demand recent changes annotations to highlight lines changed by the most recent commit
+- An much [more](https://gitlens.amod.io/#features).
+
+![Gitlens](images/gitlens-preview.gif)
+
+## [Ruby](https://marketplace.visualstudio.com/items?itemName=rebornix.Ruby)
+
+This extension provides enhanced Ruby language and debugging support, one of the coolest things about this extension is the syntax support to our ruby files, is a **must-have** if you are a Ruby developer.
+
+**Features:**
+
+- Automatic Ruby environment detection with support for rvm, rbenv, chruby, and asdf
+- Lint support via RuboCop, Standard, and Reek
+- Format support via RuboCop, Standard, Rufo, and RubyFMT
+- Semantic code folding support
+- Semantic highlighting support
+- Basic Intellisense support
+
+![VSCode Ruby](images/vsrubycode.png)
+
+## [Ruby Solargraph](https://marketplace.visualstudio.com/items?itemName=castwide.solargraph)
+
+Another **must-have** extension if you are a Ruby developer, gives Visual Studio Code some features that can be found in IDE's like "Go to definition" for classes, modules, and methods. Also provides documentation and code completion. If you are learning Ruby, this extension will help you a lot.
+
+**Features:**
+
+- Context-aware suggestions and documentation for the Ruby core
+- Detection of some variable types and method return values (e.g., String.new. returns String instance methods)
+- Identification of local, class, and instance variables within the current scope
+- Find references and rename symbols (experimental as of solargraph gem v0.22.0)
+- Support for gems
+- Near-complete support for the Ruby core and stdlib
+- Partial support for Ruby on Rails
+
+![Ruby Solargraph](images/rubysolargraph.gif)
+
+## [ruby-rubocop](https://marketplace.visualstudio.com/items?itemName=misogi.ruby-rubocop)
+
+Having rubocop configured in our ruby project will help us to maintain the quality of the code, especially if our project is being developed by a team. This extension will let us autoformat and correct our code basing it the directives written in the .rubucop.yml file. With only a key shortcut your code will be formatted in a second, is fast, is good, is a **must-have** extension.
```

Comment:
```suggestion
Having rubocop configured in your ruby project will help you to maintain the quality of the code, especially if your project is being developed by a team. This extension will let you autoformat and correct your code basing it the directives written in the .rubucop.yml file. With only a key shortcut your code will be formatted in a second, is fast, is good, is a **must-have** extension.
```

You were talking in `you` and not `us` so I think like this would keep consistency.

Reply:
`RuboCop`

Reply:
Maybe `must-have` is being used a lot haha. You could also use `essential`, `vital`, or something like that.